### PR TITLE
Restructure transports

### PR DESCRIPTION
**Pull request by sloops77**

### CODE REVIEW

**Comment by sloops77** [comment]
The aim of this restrcuture is to introduce the MessagingAdaptor (request, fireAndForget, publish, subscribe, subscribeGroup, addRequestHandler) & QueueMessagingAdaptor subclass and then the QueueTransport abstract class (process, enqueue, registerSubscriber, getSubscribers, stopProcessing) that the *existing* transports all use.

The last step of the restructure is then to change Memory and Socket comms to not implement QueueTransport but instead implement MessagingAdaptor directly - to map to their apis better

File: packages/messaging-queue/src/QueueMessagingAdaptor.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    clearTimeout(requestTimer);
+  }
+}
+
+export default class QueueMessagingAdaptor extends MessagingAdaptor {
+  constructor(transport, source, log, requestTimeout) {
+    super(log);
+    this.transport = transport;
+
+    // Subscription Stuf
```
Comment (YosBD) [inline packages/messaging-queue/src/QueueMessagingAdaptor.js]: stuff

File: packages/core/src/stellarLoaderFactory.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import forOwn from 'lodash/forOwn';
+import forEach from 'lodash/forEach';
+import isPlainObject from 'lodash/isPlainObject';
+import isArray from 'lodash/isArray';
+import head from 'lodash/head';
+
+export default function stellarLoaderFactory(stellarHandler) {
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/core/src/stellarLoaderFactory.js]: yes - but this is totally deprecated. so maybe we can live with it? until the old endpoints are removed

File: packages/core/test/StellarPubSub.test.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+      defaultPubSub.use(/.*/, mwMock);
+
+      await defaultPubSub.publish(channel, message.body);
+
+      expect(messagingMock.publish).toHaveBeenCalled();
+      expect(messagingMock.publish.mock.calls).toHaveLength(1);
+      expect(messagingMock.publish.mock.calls[0]).toEqual([channel, message]);
+
+      await Promise.delay(50);
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/core/test/StellarPubSub.test.js]: its very difficult in pubsub as the subscribe function needs to execute, and only then do we know if we are done or not

File: packages/core/src/stellarLoaderFactory.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  function _handleLoader(resource, method, value) {
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/core/src/stellarLoaderFactory.js]: deprecated - will be removed soon

File: packages/core/src/StellarCore.js [hunk 5]
```diff
@@ -123,47 +129,33 @@ class StellarCore {
...
-        match(jobData.headers.queueName || jobData.headers.channel, handlers[i].pattern)) {
-        return Promise
-          .try(() => handlers[i].fn(jobData, next, options, this.log))
-          .then(result => this._handlerResult(jobData, result))
-          .catch(error => this._handlerRejection(jobData, error));
-      }
-
-      return next();
-    };
+      const runMw = (i) => {
```
[the remark this answers is not in the document]
Comment (sloops77) [reply packages/core/src/StellarCore.js]: i tried but it became a bit unweildy. lets leave it for now

**Pull request closed without merging**